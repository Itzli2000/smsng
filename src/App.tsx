import { Grid, Typography } from "@mui/material";
import useAuthStore from "./store/authStore";
import useDataStore from "./store/dataStore";
import useFilterStore from "./store/filterStore";

function App() {
  const { isAuthenticated, user } = useAuthStore();
  const { filters } = useFilterStore();
  const { products } = useDataStore();


  return (
    <Grid container>
      <Grid item xs={12} lg={6} style={{ margin: "auto" }}>
        <Typography>isAuthenticated: {isAuthenticated}</Typography>
        <Typography>user: {JSON.stringify(user)}</Typography>
        <Typography>filters: {JSON.stringify(filters)}</Typography>
        <Typography>products: {JSON.stringify(products)}</Typography>
      </Grid>
    </Grid>
  );
}

export default App;
