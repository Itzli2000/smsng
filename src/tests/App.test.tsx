import { render, screen } from "@testing-library/react";
import App from "../App";
import { expect, test } from "vitest";

test("renders the React logo", () => {
  render(<App />);
  const isAuthenticated = screen.getByText(/isAuthenticated:/i);
  const user = screen.getByText(/user:/i);
  const filters = screen.getByText(/filters:/i);
  const products = screen.getByText(/products:/i);
  
  expect(isAuthenticated).toBeInTheDocument();
  expect(user).toBeInTheDocument();
  expect(filters).toBeInTheDocument();
  expect(products).toBeInTheDocument();
});

test("match snapshot", () => {
  const tree = render(<App />);
  expect(tree).toMatchSnapshot();
});
