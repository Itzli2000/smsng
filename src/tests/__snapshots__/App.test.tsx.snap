// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6 css-f0xsrb-MuiGrid-root"
          style="margin: auto;"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            isAuthenticated: 
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            user: 
            null
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            filters: 
            {"stores":[],"models":[]}
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            products: 
            []
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-6 css-f0xsrb-MuiGrid-root"
        style="margin: auto;"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          isAuthenticated: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          user: 
          null
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          filters: 
          {"stores":[],"models":[]}
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          products: 
          []
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
